docs/docs/use_cases/qa_structured/cube.ipynb

```diff
@@ -0,0 +1,366 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2023-11-09T09:14:23.865167Z",
+     "start_time": "2023-11-09T09:14:23.072519Z"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": "{'query': {'measures': ['ActiveUsers.wauToMau'],\n  'limit': 5,\n  'order': [{'id': 'ActiveUsers.wauToMau', 'desc': True}],\n  'rowLimit': 5,\n  'timezone': 'UTC',\n  'filters': [],\n  'dimensions': [],\n  'timeDimensions': []},\n 'data': [{'ActiveUsers.wauToMau': '100.0000000000000000'}],\n 'lastRefreshTime': '2023-11-09T09:11:43.571Z',\n 'refreshKeyValues': [],\n 'usedPreAggregations': {},\n 'transformedQuery': {'sortedDimensions': [],\n  'sortedTimeDimensions': [],\n  'timeDimensions': [],\n  'measures': ['ActiveUsers.wauToMau'],\n  'leafMeasureAdditive': False,\n  'leafMeasures': ['ActiveUsers.weeklyActiveUsers',\n   'ActiveUsers.monthlyActiveUsers'],\n  'hasNoTimeDimensionsWithoutGranularity': True,\n  'allFiltersWithinSelectedDimensions': True,\n  'isAdditive': False,\n  'granularityHierarchies': {'year': ['year',\n    'quarter',\n    'month',\n    'month',\n    'day',\n    'hour',\n    'minute',\n    'second'],\n   'quarter': ['quarter', 'month', 'day', 'hour', 'minute', 'second'],\n   'month': ['month', 'day', 'hour', 'minute', 'second'],\n   'week': ['week', 'day', 'hour', 'minute', 'second'],\n   'day': ['day', 'hour', 'minute', 'second'],\n   'hour': ['hour', 'minute', 'second'],\n   'minute': ['minute', 'second'],\n   'second': ['second']},\n  'hasMultipliedMeasures': False,\n  'hasCumulativeMeasures': True,\n  'windowGranularity': 'day',\n  'filterDimensionsSingleValueEqual': {}},\n 'requestId': 'd4592364-3a73-4419-ba0e-adccb9e18c7a-span-1',\n 'annotation': {'measures': {'ActiveUsers.wauToMau': {'title': 'Active Users WAU to MAU',\n    'shortTitle': 'WAU to MAU',\n    'type': 'number',\n    'format': 'percent',\n    'drillMembers': [],\n    'drillMembersGrouped': {'measures': [], 'dimensions': []}}},\n  'dimensions': {},\n  'segments': {},\n  'timeDimensions': {}},\n 'dataSource': 'default',\n 'dbType': 'postgres',\n 'extDbType': 'cubestore',\n 'external': False,\n 'slowQuery': False}"
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "cube.load(response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see, the Cube Query Builder chain **only created** the query, and we handled the **query execution separately**."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Get the SQL Code generated by Cube to be executed in the database.\n",
+    "\n",
+    "## Note\n",
+    "\n",
+    "`cube.sql` relies on Cube's `/v1/sql` API.\n",
+    "\n",
+    "- `/v1/sql` retrieves sql information through Cube queries [https://cube.dev/docs/reference/rest-api#v1sql](https://cube.dev/docs/reference/rest-api#v1sql)"
```

Comment:
Could you please explain why would you want to utilize `/sql` endpoint?

Reply:
Sometimes, we do not wish to directly query data; we simply want to obtain the SQL. Subsequently, there might be a need to make secondary modifications to the SQL based on certain business requirements. That's why we are adding a /sql endpoint.